ui/package.json

```diff
@@ -84,7 +84,7 @@
     "core-js": "^2.5.3",
     "d3": "^4.12.2",
     "highcharts": "^6.2.0",
-    "highcharts-react-official": "^2.0.0",
+    "highcharts-react-official": "2.0.0",
```

Comment:
Does this mean we had to downgrade highcharts to remove the vulnerabilities?

Reply:
That's just an artifact from when I tried to upgrade highcharts but that caused more issues that I determined would be outside the scope of this ticket (the upgrade didn't fix any dependencies issues).

I'll try reverting the change to `"^2.0.0"` to see if that still works.

As an aside, there are arguments for pinning the version to a single version to maintain consistency across environments (https://medium.com/the-guild/how-should-you-pin-your-npm-dependencies-and-why-2b8d545c7312)

------

my notes from upgrading highcharts-react-official:

`yarn dev-up` now fails

```
ERROR in node_modules/highcharts-react-official/dist/highcharts-react.min.d.ts(139,31): error TS2694: Namespace 'Highcharts' has no exported member 'ChartCallbackFunction'.
src/app/pages/data/cohort-review/individual-participants-charts.tsx(126,9): error TS2322: Type '{ chart: { type: string; zoomType: string; }; credits: { enabled: boolean; }; title: { text: string; }; xAxis: { title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }; yAxis: ({ ...; } | { ...; })[]; plotOptions: { ...; }; ...' is not assignable to type 'Options'.
  Types of property 'xAxis' are incompatible.
    Type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }' is not assignable to type 'AxisOptions | AxisOptions[]'.
      Type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }' is not assignable to type 'AxisOptions[]'.
        Property 'length' is missing in type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }'.
src/app/pages/data/cohort-review/review-domain-charts.tsx(77,11): error TS2322: Type '{ chart: { type: string; zoomType: string; backgroundColor: string; height: number; }; credits: { enabled: boolean; }; title: { text: string; }; yAxis: { lineWidth: number; lineColor: string; }; plotOptions: { ...; }; tooltip: { ...; }; xAxis: { ...; }; series: { ...; }[]; }' is not assignable to type 'Options'.
  Types of property 'plotOptions' are incompatible.
    Type '{ series: { label: { connectorAllowed: boolean; }; }; }' is not assignable to type 'PlotOptions'.
      Types of property 'series' are incompatible.
        Type '{ label: { connectorAllowed: boolean; }; }' has no properties in common with type 'SeriesChart'.

ℹ ｢wdm｣: Failed to compile.
```

in https://github.com/highcharts/highcharts-angular/issues/107
someone mentioned that setting the version # fixed the issue. Reverted the version number to 2.0.0:

`yarn add highcharts-react-official@2.0.0`

Reply:
upgrading to ^2.0.0 installed 2.2.2 which causes a number of test failures:


```

Summary of all failing tests
 FAIL  src/app/pages/data/concept/concept-homepage.spec.tsx (42.234s)
  ● ConceptHomepage › should default to standard concepts only, and performs a full search

    Method “text” is meant to be run on 1 node. 0 found instead.

      109 |     // Test that it switches to the table view
      110 |     expect(wrapper.find('[data-test-id="conceptTable"]').length).toBeGreaterThan(0);
    > 111 |     const firstDomainRowName = wrapper.find('[data-test-id="conceptName"]').at(1).text();
          |                                                                                   ^
      112 |     await waitOneTickAndUpdate(wrapper);
      113 |
      114 |     // Test that it changes the table when a new domain is selected

      at ReactWrapper.single (node_modules/enzyme/src/ReactWrapper.js:1166:13)
      at ReactWrapper.single [as text] (node_modules/enzyme/src/ReactWrapper.js:629:17)
      at Object.<anonymous> (src/app/pages/data/concept/concept-homepage.spec.tsx:111:83)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

  ● ConceptHomepage › should change search criteria when standard only not checked

    expect(received).toBe(expected) // Object.is equality

    Expected: 3
    Received: 1

      151 |     // check number of rows in table plus header row
      152 |     expect(wrapper.find('[data-test-id="conceptName"]').length)
    > 153 |       .toBe(conceptsCountInDomain(selectedDomain, false) + 1);
          |        ^
      154 |   });
      155 |
      156 |   it('should display the selected concepts on header', async() => {

      at Object.<anonymous> (src/app/pages/data/concept/concept-homepage.spec.tsx:153:8)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

  ● ConceptHomepage › should display the selected concepts on header

    Method “simulate” is meant to be run on 1 node. 0 found instead.

      160 |     await waitOneTickAndUpdate(wrapper);
      161 |
    > 162 |     wrapper.find('span.p-checkbox-icon.p-clickable').at(1).simulate('click');
          |                                                            ^
      163 |     await waitOneTickAndUpdate(wrapper);
      164 |     expect(wrapper.find('[data-test-id="selectedConcepts"]').text()).toBe('1');
      165 |   });

      at ReactWrapper.single (node_modules/enzyme/src/ReactWrapper.js:1166:13)
      at ReactWrapper.single [as simulate] (node_modules/enzyme/src/ReactWrapper.js:665:17)
      at Object.<anonymous> (src/app/pages/data/concept/concept-homepage.spec.tsx:162:60)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

  ● ConceptHomepage › should display the selected concepts on sliding button

    Method “simulate” is meant to be run on 1 node. 0 found instead.

      175 |       .parent().props()['disable']).toBeTruthy();
      176 |
    > 177 |     wrapper.find('span.p-checkbox-icon.p-clickable').at(1).simulate('click');
          |                                                            ^
      178 |     await waitOneTickAndUpdate(wrapper);
      179 |     expect(wrapper.find('[data-test-id="sliding-button"]')
      180 |       .parent().props()['disable']).toBeFalsy();

      at ReactWrapper.single (node_modules/enzyme/src/ReactWrapper.js:1166:13)
      at ReactWrapper.single [as simulate] (node_modules/enzyme/src/ReactWrapper.js:665:17)
      at Object.<anonymous> (src/app/pages/data/concept/concept-homepage.spec.tsx:177:60)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

  ● ConceptHomepage › should clear search and selected concepts

    Method “simulate” is meant to be run on 1 node. 0 found instead.

      188 |     await waitOneTickAndUpdate(wrapper);
      189 |
    > 190 |     wrapper.find('span.p-checkbox-icon.p-clickable').at(1).simulate('click');
          |                                                            ^
      191 |     await waitOneTickAndUpdate(wrapper);
      192 |     expect(wrapper.find('[data-test-id="selectedConcepts"]').text()).toBe('1');
      193 |

      at ReactWrapper.single (node_modules/enzyme/src/ReactWrapper.js:1166:13)
      at ReactWrapper.single [as simulate] (node_modules/enzyme/src/ReactWrapper.js:665:17)
      at Object.<anonymous> (src/app/pages/data/concept/concept-homepage.spec.tsx:190:60)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

 FAIL  src/app/pages/data/concept/concept-set-details.spec.tsx (44.51s)
  ● ConceptSetDetails › should remove concepts

    expect(received).toEqual(expected)

    Expected value to equal:
      5
    Received:
      6

      126 |     await waitOneTickAndUpdate(wrapper);
      127 |     expect(wrapper.find('span.p-checkbox-icon.p-clickable').length)
    > 128 |       .toEqual(numConceptsPlusHeader - 1);
          |        ^
      129 |   });
      130 |
      131 |   // TODO RW-2625: test edit and delete set from popup trigger menu

      at Object.<anonymous> (src/app/pages/data/concept/concept-set-details.spec.tsx:128:8)
      at step (node_modules/tslib/tslib.js:136:27)
      at Object.next (node_modules/tslib/tslib.js:117:57)
      at fulfilled (node_modules/tslib/tslib.js:107:62)

 FAIL  src/app/pages/data/cohort-review/individual-participants-charts.spec.tsx
  ● Test suite failed to run

    TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    src/app/pages/data/cohort-review/individual-participants-charts.tsx:126:9 - error TS2322: Type '{ chart: { type: string; zoomType: string; }; credits: { enabled: boolean; }; title: { text: string; }; xAxis: { title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }; yAxis: ({ ...; } | { ...; })[]; plotOptions: { ...; }; ...' is not assignable to type 'Options'.
      Types of property 'xAxis' are incompatible.
        Type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }' is not assignable to type 'AxisOptions | AxisOptions[]'.
          Type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }' is not assignable to type 'AxisOptions[]'.
            Property 'length' is missing in type '{ title: { enabled: boolean; text: string; }; labels: { formatter: () => string; }; startOnTick: boolean; endOnTick: boolean; tickInterval: number; }'.

    126         options={this.getOptions()}
                ~~~~~~~

      node_modules/highcharts-react-official/dist/highcharts-react.min.d.ts:122:9
        122         options?: Highcharts.Options;
                    ~~~~~~~
        The expected type comes from property 'options' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<HighchartsReact> & Readonly<Props> & Readonly<{ children?: ReactNode; }>'


Test Suites: 3 failed, 55 passed, 58 total
Tests:       6 failed, 194 passed, 200 total
Snapshots:   0 total
Time:        123.99s
Ran all test suites.
error Command failed with exit code 1.
```

Reply:
I see issues crop up even when updating to 2.1.0 so I think we should pin it at 2.0.0